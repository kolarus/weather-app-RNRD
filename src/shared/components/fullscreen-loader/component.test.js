import React from 'react';
import {render} from '@testing-library/react-native';

import FullscreenLoader from './component';

test('FullscreenLoader should match snapshot', () => {
  const {toJSON} = render(<FullscreenLoader isLoading={true} description="Test Descr" />);

  expect(toJSON()).toMatchSnapshot();
});

test('FullscreenLoader should have passed description text if loading is true', () => {
  const {getByText} = render(<FullscreenLoader isLoading={true} description="Test Descr" />);

  expect(getByText('Test Descr')).not.toBeNull();
});

test('FullscreenLoader should not have passed description text if loading is false', () => {
  const {queryByText} = render(<FullscreenLoader isLoading={false} description="Test Descr" />);

  expect(queryByText('Test Descr')).toBeNull();
});
