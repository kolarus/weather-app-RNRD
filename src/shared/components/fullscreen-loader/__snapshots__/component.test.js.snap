// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullscreenLoader should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "gray",
      "bottom": 0,
      "justifyContent": "flex-end",
      "left": 0,
      "opacity": 0,
      "paddingBottom": "30%",
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "PT Sans",
          "fontSize": 15,
        },
        undefined,
        Object {},
      ]
    }
  />
</View>
`;
